curriculum/challenges/english/25-front-end-development/quiz-typescript/66f1ae758b77cfc3e4da6151.md

```diff
@@ -17,439 +17,454 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --distractors--
 
-Placeholder distractor 1
+Java
 
 ---
 
-Placeholder distractor 2
+Python
 
 ---
 
-Placeholder distractor 3
+C++
 
 #### --answer--
 
-Placeholder answer
+JavaScript
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What would be the output of the following code snippet?
+
+```ts
+function sum(a: number, b: number): number {
+  return a + b;
+}
+console.log(sum(2, 3));
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+`2 + 3`
 
 ---
 
-Placeholder distractor 2
+`undefined`
 
 ---
 
-Placeholder distractor 3
+`Error: Type 'number' is not assignable to type 'string'`
 
 #### --answer--
 
-Placeholder answer
+`5`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the primary purpose of an interface in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+To compile TypeScript code into JavaScript.
 
 ---
 
-Placeholder distractor 2
+To create a new data type.
 
 ---
 
-Placeholder distractor 3
+To create a new JavaScript Object.
 
 #### --answer--
 
-Placeholder answer
+To define the structure of an object.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is NOT a basic primitive in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+string
 
 ---
 
-Placeholder distractor 2
+boolean
 
 ---
 
-Placeholder distractor 3
+number
 
 #### --answer--
 
-Placeholder answer
+array
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following correctly defines a literal type in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+`let y: string;`
 
 ---
 
-Placeholder distractor 2
+`let z: boolean;`
 
 ---
 
-Placeholder distractor 3
+`let a: number[];`
 
 #### --answer--
 
-Placeholder answer
+`let x: 5;`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the primary purpose of the `type` keyword in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+To declare variables
 
 ---
 
-Placeholder distractor 2
+To create classes
 
 ---
 
-Placeholder distractor 3
+To implement interfaces
 
 #### --answer--
 
-Placeholder answer
+To define custom types
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following correctly defines a union type?
 
 #### --distractors--
 
-Placeholder distractor 1
+`type Result = Pass && Fail;`
 
 ---
 
-Placeholder distractor 2
+`type Result = Pass & Fail;`
 
 ---
 
-Placeholder distractor 3
+`type Result = Pass + Fail;`
 
 #### --answer--
 
-Placeholder answer
+`type Result = Pass | Fail;`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following capabilities is unique to `interface` and CANNOT be achieved using `type` in TypeScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+Defining object shapes
 
 ---
 
-Placeholder distractor 2
+Creating union types
 
 ---
 
-Placeholder distractor 3
+Creating intersection types
 
 #### --answer--
 
-Placeholder answer
+Implementing a class
```

Comment:
Not exactly sure what is meant by, implementing a class?
If it is referring to the use of `extends` could we be more explicit about it?

Reply:
Thanks for pointing it out. I should've phrased that better.
What I meant was that classes can implement interfaces or Interfaces can be used to declare class implementations.

Thanks again